ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +71,50 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
-    top: Math.floor(heightWithMargins * pageIndex + marginTopPixels + (height - bottomPixels)),
+  const pageId = generatePageIdFromIndex(pageIndex);
+
+  const pageIdElement = document.getElementById(pageId);
+
+  if (!pageIdElement) {
+    return;
+  }
+
+  const parentElement = getScrollParent(pageIdElement);
+
+  if (!parentElement) {
+    return;
+  }
+
+  parentElement.scrollTo({
+    top: calculateTop(heightWithMargins, pageIndex, marginTopPixels, height, bottomPixels),
     left: Math.floor(leftPixels),
     behavior: 'smooth',
   });
 }
 
+export function getScrollParent(node: HTMLElement): HTMLElement | null {
+  if (typeof document === 'undefined') {
+    return null;
+  }
+  if (!node) {
+    return document.documentElement;
+  }
+  if (node.scrollHeight > node.clientHeight) {
+    return node;
+  }
+  return getScrollParent(node.parentNode as HTMLElement);
```

Comment:
You might be able to get away with `node.parentElement`. IE doesn't like is...but we don't like IE...so it's mutual. 😆